packages/color-mode/src/color-mode-script.tsx

```diff
@@ -1,63 +1,35 @@
-import { ColorMode, ConfigColorMode } from "./color-mode-provider"
-
-export function setScript(initialValue: ConfigColorMode) {
-  const mql = window.matchMedia("(prefers-color-scheme: dark)")
-  const systemPreference = mql.matches ? "dark" : "light"
-
-  let persistedPreference: ColorMode = systemPreference
-
-  try {
-    persistedPreference = localStorage.getItem(
-      "chakra-ui-color-mode",
-    ) as ColorMode
-  } catch (error) {
-    console.log(
-      "Chakra UI: localStorage is not available. Color mode persistence might not work as expected",
-    )
-  }
-
-  let colorMode: ColorMode
-
-  if (persistedPreference) {
-    colorMode = persistedPreference
-  } else if (initialValue === "system") {
-    colorMode = systemPreference
-  } else {
-    colorMode = initialValue ?? systemPreference
-  }
-
-  if (colorMode) {
-    /**
-     * Keep in sync with `root.set() {@file ./color-mode.utils.ts}
-     */
-    document.documentElement.setAttribute("data-theme", colorMode)
-  }
-}
 
-interface ColorModeScriptProps {
-  initialColorMode?: ConfigColorMode
-  /**
-   * Optional nonce that will be passed to the created `<script>` tag.
-   */
-  nonce?: string
+export type ColorModeScriptProps = {
+  type?: "localStorage" | "cookie"
+  initialColorMode?: "light" | "dark" | "system"
+  storageKey?: string
 }
 
-/**
- * Script to add to the root of your application when using localStorage,
- * to help prevent flash of color mode that can happen during page load.
- */
-export const ColorModeScript = (props: ColorModeScriptProps) => {
-  let { initialColorMode = "light" } = props
+export function getScriptSrc(props: ColorModeScriptProps = {}) {
+  const {
+    initialColorMode = "system",
+    type = "localStorage",
+    storageKey = "chakra-ui-color-mode",
+  } = props
+
+  const isCookie = type === "cookie"
+
+  const init = isCookie ? initialColorMode : `'${initialColorMode}'`
+  const k = isCookie ? storageKey : `'${storageKey}'`
 
-  // Runtime safeguard against invalid color mode values.
-  const validColorModeValues = ["dark", "light", "system"] as const
-  if (!validColorModeValues.includes(initialColorMode)) {
-    initialColorMode = "light"
-  }
+  const cookieScript = `!function(){try{var t="(prefers-color-scheme: dark)", e=window.matchMedia(t).matches?"dark":"light", d=document.documentElement;var ck=document.cookie.match(new RegExp(\`(^| )${k}=([^;]+)\`));var m=ck && ck[2]; if(!m) return document.cookie=\`${k}=${init}; max-age=31536000; path=/\`,d.dataset.theme="system"==='${init}'?e:'${init}';var yy="system"===m?e:m; d.dataset.theme=yy;d.style.colorScheme=yy}catch(t){}}()`
+
+  const localStorageScript = `!function(){try{var t="(prefers-color-scheme: dark)",e=window.matchMedia(t).matches?"dark":"light",d=document.documentElement,m=localStorage.getItem(${k});if(!m)return localStorage.setItem(${k},${init}),d.dataset.theme="system"===${init}?e:${init};var yy="system"===m?e:m; d.dataset.theme=yy;d.style.colorScheme=yy}catch(t){}}()`
```

Comment:
Looks interesting, but seems better than calling `toString()` on a function to get the source.

Reply:
My aim with this function is to really slim down the script given that it won't change that often compared to the rest of the codebase. Using the shebang `!function(){}()` also ensures that browser executes the script immediately.